File: DemoParser.Tests/RegionParserTests/Go/Input.go
//region Usings
import { 
   "github.com/ravendb/ravendb-go-client"
   "encoding/json"
}  
//endregion
  
//region Demo
func createDocument(companyName, companyPhone, contactName, contactTitle string) error {

    //region Step_1
    newCompany := Company {
        Name:  companyName,
        Phone: companyPhone,
        Contact: &Contact {
          Name:  contactName,
          Title: contactTitle,
         },
    }
    //endregion

    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()

    //region Step_2
    err = session.Store(newCompany)
    if err != nil {
        return err
    }
    //endregion

    //region Step_3
    theNewDocumentID := newCompany.ID
    //endregion

    //region Step_4
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    //endregion

    return nil
}
//endregion

File: DemoParser.Tests/RegionParserTests/NodeJs/input.js
/* eslint-disable no-unused-vars */
//region Usings
//endregion
const { store } = require('../../common/docStoreHolder');
const { Company, Contact } = require('../../common/models');

async function run ({ companyName, companyPhone, contactName, contactTitle }) {
    //region Demo
    //region Step_1
    const newCompany = new Company();
    newCompany.name = companyName;
    newCompany.phone = companyPhone;

    const newContact = new Contact();
    newContact.name = contactName;
    newContact.title = contactTitle;

    newCompany.contact = newContact;
    //endregion

    const session = store.openSession();
    //region Step_2
    await session.store(newCompany);
    //endregion

    //region Step_3
    const theNewDocumentId = newCompany.id;
    //endregion

    //region Step_4
    await session.saveChanges();
    //endregion
    //endregion
}

module.exports = { run };
